staging/src/k8s.io/apiserver/pkg/server/options/etcd.go

```diff
@@ -291,7 +342,11 @@ func (s *EtcdOptions) addEtcdHealthEndpoint(c *server.Config) error {
 		return readyCheck()
 	}))
 
-	c.AddHealthChecks(s.kmsPluginHealthzChecks...)
+	if s.EncryptionProviderConfigAutomaticReload {
```

Comment:
see above comments on making `c.AddHealthChecks(s.kmsPluginHealthzChecks...)` correct whether we are reloading or not